app_integrations/apps/onelogin.py

```diff
@@ -124,59 +128,50 @@ def _get_onelogin_events(self):
                 }
             ]
         """
-        # OneLogin API expects the ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ
-        formatted_date = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
-        params = {'since': formatted_date}
-
         # Make sure we have authentication headers
         if not self._auth_headers:
             return False
 
-        events = self._get_onelogin_paginated_events(params)
-
-        while self._more_to_poll and self._next_page_url:
-            LOGGER.debug('More events to retrieve for \'%s\': %s', self.type(), self._more_to_poll)
-            pagination = self._get_onelogin_paginated_events(None)
+        # Are we just getting events or getting paginated events?
+        if self._next_page_url:
+            params = None
+        else:
+            # OneLogin API expects the ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ
+            formatted_date = datetime.fromtimestamp(
+                self._last_timestamp).strftime('%Y-%m-%dT%H:%M:%SZ')
+            params = {'since': formatted_date}
 
-            # Add the events to our results, this is equivalent to using events.extend()
-            events += pagination
+        LOGGER.debug('Events to retrieve for \'%s\': %s', self.type(), self._more_to_poll)
+        response = self._make_get_request(self._next_page_url, self._auth_headers, params)
 
-        # Return the list of logs to the caller so they can be send to the batcher
-        return events
+        if not response:
+            events = False
```

Comment:
can we just `return False` here instead? the line below (151) could result in a KeyError